spec/registry.xml

```diff
@@ -0,0 +1,250 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-model href="registry.dtd" type="application/xml-dtd"?>
+<registry>
+  <pattern id="anyNumber" regex="-?[0-9]+(\.[0-9]+)"/>
+  <pattern id="positiveInteger" regex="[0-9]+"/>
+  <pattern id="currencyCode" regex="[A-Z]{3}"/>
+  <pattern id="timeZoneId" regex="[a-zA-Z/]+"/>
+  <pattern id="anythingNotEmpty" regex=".+"/>
+
+  <!-- I don't think that regexp is powerfull enough to
+    describe all things we might want, especially for input.
+    ISO 8601 is one example.
+    https://en.wikipedia.org/wiki/ISO_8601
+    So for now this is a very "sketchy" regexp.
+
+    One can also imagine phone numbers, emails, url, etc.
+
+    Even the regex for anyNumber (from the .md) is not OK:
+      .3  : Invalid (no digit before dot)
+      3   : Invalid (no dot + one digit after)
+            Brackets in regexp means grouping, not optional.
+      002 : Valid (OK to start with zero)
+      000 : Valid (OK to start with zero)
+  -->
+  <pattern id="iso8601" regex="\d{4}-\d{2}-\d{2}T\d{2}:\d{2}\d{2}"/>
+
+  <function name="datetime">
+    <!-- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat -->
+    <description>Language-sensitive date and time formatting</description>
+
+    <formatSignature>
+      <input pattern="iso8601"/>
+      <!-- The predefined date formatting style to use. -->
+      <option name="dateStyle" values="full long medium short"/>
```

Comment:
should the list be in the smallest-to-largest order?

Reply:
Following the ECMA order:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#datestyle